lua/weapons/weapon_funnybanana.lua

```diff
@@ -0,0 +1,135 @@
+SWEP.PrintName = "Funny Picture of a Banana"
+SWEP.Purpose = "A hilarious picture. Look at it!"
+SWEP.Author = "John J. Callanan"
+SWEP.Instructions = "Left Click: Laugh\nRight Click: Laugh Hard"
+
+if CLIENT then
+	SWEP.WepSelectIcon = surface.GetTextureID("chev/weapon_funnybanana/funnybananawepicon")
+	SWEP.BounceWeaponIcon = false
+end
+
+SWEP.ViewModel = "models/chev/bananaframe.mdl"
+SWEP.WorldModel = "models/chev/bananaframe.mdl"
+
+function SWEP:PrimaryAttack()
+	if CLIENT then
+		RunConsoleCommand("act", "laugh")
+		
+		if math.random(0, 10) < 3 then --random chance to banana-ify the screen
+			RunConsoleCommand("pp_texturize", "pp/texturize/banana.png")
+			timer.Simple(6, function()
+				RunConsoleCommand("pp_texturize", "")
+			end)
+		end
+	end
+	if SERVER then
+		local cartoonsnd = {"funnysounds01.ogg", "funnysounds02.ogg"}
+
+		self.Owner:Say("hahaha! what a funny picture! ;banana;")
+		self.Owner:ExtEmitSound("weapon_funnybanana/hahaha_funnypicture.ogg", {shared=true, level=70, channel=CHAN_WEAPON})
+
+		for k, v in pairs(player.GetAll()) do
+			if self.Owner:GetPos():Distance(v:GetPos()) < 200 then
+				if v != self.Owner then
+					v:ExtEmitSound("weapon_funnybanana/hahaha.ogg", {shared=true, level=70})
+				end
+			end
+		end
+		
+		timer.Simple(2, function()
+			if IsValid(self) and IsValid(self.Owner) then
+				self.Owner:ExtEmitSound("weapon_funnybanana/audiencelaugh.ogg", {shared=true, level=65, volume=0.7})
+				self.Owner:ExtEmitSound("weapon_funnybanana/slipsoundc.ogg", {shared=true, level=65, volume=0.5})
+				self.Owner:ExtEmitSound("weapon_funnybanana/"..cartoonsnd[math.random(#cartoonsnd)], {shared=true, level=65, volume=0.4})
+				self.Owner:ExtEmitSound("airhorn/honk1.ogg", {shared=true, level=65, volume=0.5})
+			end
+		end)
+	end
+
+	self:SetNextPrimaryFire(CurTime() + 10)
+end
+
+function SWEP:SecondaryAttack() --Same as primary attack, but you laugh so hard you die (because the picture is very funny)
```

Comment:
This doesn't have a cool down, you can spam right click

Reply:
This still doesn't totally fix it, the issue now is just that primary and secondary fire have separate cool downs. Good enough though